### PR TITLE
[Docs/PTX] Add device tensor map init example

### PR DESCRIPTION
**Pull request by ahendriksen**
## Description

closes https://github.com/NVIDIA/cccl/issues/1982

Adds documentation of tensormap modification/initialization on device. Also improves on navigation and table of contents of the cuda::ptx docs.

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: docs/libcudacxx/ptx/examples/device_tensormap_initialization.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+The ``get_cuTensorMapEncodeTiled`` function is omitted from the code example. It
+can be found in the CUDA Programming Guide or in the full code example linked
+below.
+
+Full code example
+-----------------
+
+The full code example is included below. The `cuda::ptx` instructions in this
+tutorial have become available in CUDA Toolkit 12.5.
```
Comment (ahendriksen) [inline docs/libcudacxx/ptx/examples/device_tensormap_initialization.rst:138]: Godbolt does not yet have CUDA 12.5, so there is no link here. I didn't want to link to a broken example.

File: docs/libcudacxx/ptx/examples/device_tensormap_initialization.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+1. Pass an existing tensor map, the template, to the kernel. In contrast to
+   kernels that ***use** the tensor map in a ``cp.async.bulk.tensor`` instruction,
+   this may be done in any way: a pointer to global memory, kernel parameter, a
+   ``__const___`` variable, etc.
+
+2. Copy the template tensor map to shared memory.
+
+3. Modify the tensor map in shared memory using :ref:`tensormap.replace
+   <libcudacxx-ptx-instructions-tensormap-replace>`. This instruction can be
+   used to modify any field of the tensor map, including the base addres, size,
```
Comment (daniellustig) [inline docs/libcudacxx/ptx/examples/device_tensormap_initialization.rst:60]: address

**Review by wmaxey — APPROVED** [review]

**Review by miscco — APPROVED** [review]

**Comment by ahendriksen** [comment]
Please don't merge yet. I still have to incorporate some internal feedback.

File: docs/libcudacxx/ptx/examples/device_tensormap_initialization.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
+.. _libcudacxx-ptx-examples-device-tensormap-initialization:
+
+How to initialize a tensor map on device
+========================================
+
+An introduction to TMA (Tensor Memory Access) can be found in the `CUDA
+Programming Guide
+<https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#asynchronous-data-copies-using-tensor-memory-access-tma>`__.
```
Comment (gonidelis) [inline docs/libcudacxx/ptx/examples/device_tensormap_initialization.rst:8]:
```suggestion
<https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html?highlight=tma#asynchronous-data-copies-using-the-tensor-memory-accelerator-tma>`__.
```

**Comment by ahendriksen** [comment]
In the mean time, the code example for on-device tensor map modification has made it into the CUDA programming guide. Instead of duplicating the documentation and code sample, I have linked to the relevant section in the programming guide. The improvements to table of contents and layout are still very much worth it in this PR. 

I have copied over the fixed links from Bryan van de Ven's PR.

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**